Scripts/run-tests.sh

```diff
@@ -32,9 +36,38 @@ schemes=(
 	"VisualRecognitionV3"
 )
 
-for scheme in ${schemes[@]}; do
-	xcodebuild -scheme "$scheme" -destination "$DESTINATION" | xcpretty
-	checkStatus $?
+####################
+# Dependencies
+####################
+
+brew update
+brew outdated carthage || brew upgrade carthage
+carthage bootstrap --platform iOS
+
+####################
+# Build and Test
+####################
+
+# set a pipeline's return status to the value of the last (rightmost) commmand
+# to exit with a non-zero status, or zero if all commands exited successfully
+# (required to check status code when using xcpretty)
+set -o pipefail
+
+# build each scheme
+for SCHEME in ${SCHEMES[@]}; do
+	xcodebuild -scheme "$SCHEME" -destination "$DESTINATION" | xcpretty
+	EXIT_CODES+=($?)
 done
 
-exit $exitCode
+####################
+# Set Exit Code
+####################
+
+# exit with the first non-zero status or zero if all builds exited successfully
+for EXIT_CODE in ${EXIT_CODES[@]}; do
+	if [ $EXIT_CODE -ne 0 ]
+	then
+		exit $EXIT_CODE
+	fi
+done
```

Comment:
While this works, there is something that bugs me about running another loop to reduce the `EXIT_CODES` array to a single value, when we built up `EXIT_CODES` in the loop right above.  Why not keep a single `EXIT_CODE` value, initialized to zero before the first loop, and update it within the loop if it is still 0?  Then you can dispense with the second loop altogether and simply `exit $EXIT_CODE`.

Reply:
Leaving a note for future reference.

Mike also suggested using `||` if we are not wed to the _last_ non-zero exit code:

```bash
xcodebuild -scheme "$SCHEME" -destination "$DESTINATION" | xcpretty || EXIT_CODE=$?
```